Doc: Readme updates to align with new Red Hat template (#693)
aristanetworks/ansible-cvp: ansible_collections/arista/cvp/README.md

  ~ that can be found in the LICENSE file.
  -->

# arista.cvp - Automate CloudVision with Ansible

## Description

[Arista Networks](https://www.arista.com/) supports Ansible for managing devices running the EOS operating system through [CloudVision platform (CVP)](https://www.arista.com/en/products/eos/eos-cloudvision). This collection includes a set of Ansible modules that perform specific configuration tasks on a CVP server. These tasks include collecting facts, managing configlets, building topology with containers and devices, and running tasks.

Even if the `arista.cvp` collection is integrated with the `arista.avd` collection to [automate configuration deployment](https://avd.arista.com/4.7/roles/eos_config_deploy_cvp/index.html), this collection can also be used outside of AVD tasks to populate your CloudVision server with your workflows.

## Requirements

The CVP collection has the following requirements:

- Python 3.9+

> - Ansible Core 2.14.0 to 2.16.x

Reviewer: ```suggestion
- Ansible Core 2.15.0 to 2.17.x
```

- Install the arista.cvp collection
- [Additional Python packages](#additional-python-dependencies)

For a full breakdown of the requirements, please see the official `arista.cvp` [documentation](https://cvp.avd.sh/en/stable/docs/installation/requirements/).

## Installations

Before using this collection, you need to install it with the Ansible Galaxy command-line tool:

```shell
ansible-galaxy collection install arista.cvp
```

You can also include it in a requirements.yml file and install it with ansible-galaxy collection install -r requirements.yml, using the format:

```yaml
collections:
  - name: arista.cvp
```

Note that if you install the collection from Ansible Galaxy, it will not be upgraded automatically when you upgrade the Ansible package.

To upgrade the collection to the latest available version, run the following command:

```shell
ansible-galaxy collection install arista.cvp --upgrade
```

> You can also install a specific version of the collection, for example, if you need to downgrade when something is broken in the latest version (please report an issue in this repository). Use the following syntax to install version 4.7.1:

Reviewer: ```suggestion
You can also install a specific version of the collection, for example, if you need to downgrade when something is broken in the latest version (please report an issue in this repository). Use the following syntax to install version 3.10.1:
```

As modules of this collection are based on [`HTTPAPI` connection plugin](https://docs.ansible.com/ansible/latest/plugins/httpapi.html), authentication elements shall be declared using this plugin mechanism and are automatically shared with `arista.cvp.cv_*` modules.

## Testing

Please see our [pipeline workflows](https://github.com/aristanetworks/ansible-cvp/actions) to view the in-depth testing performed against the collection.

## Contributing

Contributing pull requests are gladly welcomed for this repository. If you are planning a significant change, please start a discussion first to ensure we can merge it.

You can also open an [issue](https://github.com/aristanetworks/ansible-cvp/issues) to report any problem or to submit enhancement.

> 

Reviewer: ```suggestion
```

## Support

> Support for this `arista.cvp` collection is provided by the community directly in this repository. If you have any questions, please open an [issue](https://github.com/aristanetworks/ansible-cvp/issues).

Reviewer: ```suggestion
- The `arista.avd` Ansible collection version 4.x releases and higher offer full support from Arista TAC, which includes the `arista.cvp` Ansible collection. If your organization has the [A-Care subscription](https://www.arista.com/assets/data/pdf/AVD-A-Care-TAC-Support-Overview.pdf) please don't hesitate to contact TAC with any questions or issues.
- Community support is provided via [Github discussions board](https://github.com/aristanetworks/ansible-cvp/discussions).
```

As modules of this collection are based on [`HTTPAPI` connection plugin](https://docs.ansible.com/ansible/latest/plugins/httpapi.html), authentication elements shall be declared using this plugin mechanism and are automatically shared with `arista.cvp.cv_*` modules.

## Testing

Please see our [pipeline workflows](https://github.com/aristanetworks/ansible-cvp/actions) to view the in-depth testing performed against the collection.

## Contributing

Contributing pull requests are gladly welcomed for this repository. If you are planning a significant change, please start a discussion first to ensure we can merge it.

You can also open an [issue](https://github.com/aristanetworks/ansible-cvp/issues) to report any problem or to submit enhancement.

> A more complete [guide for contribution](hhttps://avd.arista.com/stable/docs/contribution/overview.html) is available.

Reviewer: ```suggestion
```
